pulsar/consumer_test.go

```diff
@@ -1262,8 +1262,9 @@ func TestConsumerSeek(t *testing.T) {
 	defer producer.Close()
 
 	consumer, err := client.Subscribe(ConsumerOptions{
-		Topic:            topicName,
-		SubscriptionName: "sub-1",
+		Topic:                   topicName,
+		SubscriptionName:        "sub-1",
+		StartMessageIDInclusive: true,
```

Comment:
This change seems irrelevant?

Reply:
This is necessary, Because this test seeks to N−5 and expects to receive this message next time.

In fact, StartMessageIDInclusive should also apply to the seek operation.

Reply:
Anyway, this is a breaking change. Though the existing behavior might not be expected, users might already depend on this behavior.

Unlike `hasNext()`, the existing behavior is that if a consumer seeks to the message id, it will still consume that message. While this PR's main purpose is to resolve the wrong result of `hasNext()`. I think you should use a different PR for changing the `seek` behavior.

Reply:
Unlike `hasNext()`, if it returns true and there is no more message, the application will be stuck at `Next`.

However, the existing behavior of `seek` won't cause such serious result. Indeed, some users might think it's a special behavior of Pulsar Go client and set the message id with the inclusive semantics.

Reply:
I do not agree that this is a break change; it is more like a behavior fix or bug fix.

After this PR, it will keep the same behavior with the java client. Refer to(peek to id-5 and receive msg from id-6):
https://github.com/apache/pulsar/blob/a1f3322ed358ab6841f0d3e43f2afcc54788b887/pulsar-broker/src/test/java/org/apache/pulsar/broker/service/SubscriptionSeekTest.java#L142-L165